src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -163,7 +166,16 @@ class ClusterMonitorActor(val cluster: Cluster,
       // Only happens if the cluster was NOT created with the pet service account.
       removeServiceAccountKey
     ))
-
+    dbRef.inTransaction { dataAccess =>
```

Comment:
This future is racy, it needs to be hooked to some other future via `flatMap` or `sequence`